### PR TITLE
Add FAQ about Citations

### PR DESCRIPTION
**Pull request by iHiD**
Closes https://github.com/exercism/v3/issues/163

I'm not sure if this should go anywhere else as well?

### CODE REVIEW

**Comment by SaschaMann** [comment]
> I'm not sure if this should go anywhere else as well?

Perhaps the `[MAINTAINERS] Add new reference document` issue template?

File: docs/maintainers/faqs.md [hunk 1]
```diff
@@ -152,3 +162,4 @@ Do you have more questions or feel that something is missing? Please let us know
 [concepts]: ../../reference/README.md
 [concept-exercise-issue-example]: https://github.com/exercism/v3/issues/33
 [open-github-issue]: https://github.com/exercism/v3/issues/new/choose
+[socp-citation-style] https://paperpile.com/s/science-of-computer-programming-citation-style/#examples-website-citations
```
Comment (SaschaMann) [inline docs/maintainers/faqs.md]:
```suggestion
[socp-citation-style]: https://paperpile.com/s/science-of-computer-programming-citation-style/#examples-website-citations
```

File: docs/maintainers/faqs.md [hunk 2]
```diff
@@ -131,6 +131,16 @@ Short answer: no. For the moment, we have all the v3 tracks in this single monor
...
 The basic communication channel for maintainers is Slack, and in particular the maintainer-only `#maintaining-<track>` channel. For communication with everyone else, GitHub issues should be used, for obvious reasons (discoverability, familiarity, etc.). We've also enabled the GitHub Projects feature on the v3 repo, so you can organize your track's v3 transition there too.
 
+## Should I use citations
+
+Yes! Please use citations in any reference documentation (files under `**/reference/**`). We are using the [Science of Computer Programming citation style][socp-citation-style]. This means that when citing website, please use this format:
+
+> $TITLE, $SOURCE. ($PUBLISHED_DATE). <$URL> (accessed $ACCESSED_DATE).
+
+For example:
+> Actor model, Wikipedia. (2020). <https://en.wikipedia.org/w/index.php?title=Actor_model&oldid=939106706> (accessed February 29, 2020).
```
Comment (SaschaMann) [inline docs/maintainers/faqs.md]:
```suggestion
> $TITLE, $SOURCE. ($PUBLISHED_DATE). $URL (accessed $ACCESSED_DATE).

For example:
> Actor model, Wikipedia. (2020). https://en.wikipedia.org/w/index.php?title=Actor_model&oldid=939106706 (accessed February 29, 2020).
```

The `<>` are not part of it, according to your source and Zotero's citation exporter.

Comment (kotp) [reply docs/maintainers/faqs.md]: Markdown specification states: "The link URL may, optionally, be surrounded by angle brackets"

In doing so, it should link and show itself as the link.  Perhaps this is why it is there.  And easier to put into place from the tooling, when dropping it into markdown, than doing the \[url](url) dance.

Comment (SaschaMann) [reply docs/maintainers/faqs.md]: It shows as link without them too (at least on GitHub), and manually adding `<>` whenever citing something is rather tedious.

Comment (yawpitch) [reply docs/maintainers/faqs.md]: > It shows as link without them too (at least on GitHub), and manually adding `<>` whenever citing something is rather tedious.

Be a wee bit careful relying on Github's behavior there, I've seen quite a few situations in the wild where other markdown renderers (notably CommonMark-compliant ones) have left "naked" links entirely un-clickable.

Comment (SaschaMann) [reply docs/maintainers/faqs.md]: Ok, I found a way to adapt the style to auto-generate the citation with `<>`, so I'm fine with this.

Comment (kotp) [reply docs/maintainers/faqs.md]: Yeah, I think that this totally depends on what tooling Exercism is using...

**Comment by kotp** [comment]
But I want to bring attention to the angle brackets around the URL. I would tend to do this to signal to anyone that might want to maintain the markdown file that this is purposefully not done with what might be considered "normal" markdown of a `[title/display](url)` form. So a communications device that doesn't get in the way of display.

**Comment by iHiD** [comment]
I'll let @ErikSchierboom finish this off when he's back next week :)

**Comment by kotp** [comment]
> Can we please mention an online tool to generate this?

Do you mean Zotero?  Perhaps online, but doesn't have to be online.

**Comment by SaschaMann** [comment]
https://zbib.org/ can be used, though you have to add the `<>` and convert the numbering to a markdown list manually.

**Comment by SleeplessByte** [comment]
> Do you mean Zotero? Perhaps online, but doesn't have to be online.

Doesn't _HAVE_ to be, but makes adoption much easier.

> https://zbib.org/ can be used, though you have to add the <> and convert the numbering to a markdown list manually.

Can we write this as "instructions" so we can copy it to CONTRIBUTING?

File: docs/maintainers/faqs.md (hunk 2 above)
Comment (ErikSchierboom) [reply docs/maintainers/faqs.md]: I was also confused about the `<>` bit. I'd much prefer to use the exact format used on the website we link to.

@kotp The tooling we're using for the website is [docsify](https://github.com/exercism/v3/blob/master/docs/maintainers/website.md). I've just checked if the variant that does not use `<>` characters works there, and it does! So my suggestion would be to apply @SaschaMann's suggestion and remove these `<>` characters.

**Comment by ErikSchierboom** [comment]
> EDIT: the angle brackets won't render and I'm pretty certain it will break some non-compliant mark down renderers. Is there a particular reason we want those?

I'm also in favor of losing them (see [my comment](https://github.com/exercism/v3/pull/868#discussion_r390189293)).

I've used the https://zbib.org/ website, and it also adds quotes around the title:

> “Actor Model”. Wikipedia, 4 februari 2020. Wikipedia, https://en.wikipedia.org/w/index.php?title=Actor_model&oldid=939106706.

This is not in the specification, although I don't mind it being there.

**Comment by SaschaMann** [comment]
> I've used the https://zbib.org/ website, and it also adds quotes around the title:

You need to select the "Science of Computer Programming" bibliography style once to get the exact format as in https://github.com/exercism/v3/issues/163#issuecomment-595006151. It'll remember the style choice for future citations, I think.

**Comment by ErikSchierboom** [comment]
Ah great, thanks for explaining!

**Comment by ErikSchierboom** [comment]
I've updated the FAQ and the existing citations. Let me know what you think.

File: reference/concepts/actor_model.md [hunk 3]
```diff
@@ -1,7 +1,7 @@
 # Actor Model
 
-> The _actor model_ is a computational model of concurrency where an _actor_ is the computational primitive.  In response to messages sent by other _actors_, the _actor_ can: make local decisions, create more actors, send more messages, determine how to respond to the next message.  They can manage their own private state, and can affect other _actors_ only by sending a message.
+> The _actor model_ is a computational model of concurrency where an _actor_ is the computational primitive. In response to messages sent by other _actors_, the _actor_ can: make local decisions, create more actors, send more messages, determine how to respond to the next message. They can manage their own private state, and can affect other _actors_ only by sending a message.<sup>1</sup>
 
-## Sources
+---
 
-1. Actor model, Wikipedia. (2020). <https://en.wikipedia.org/w/index.php?title=Actor_model&oldid=939106706> (accessed February 29, 2020).
+[1] Actor model, Wikipedia. (2020). <https://en.wikipedia.org/w/index.php?title=Actor_model&oldid=939106706> (accessed February 29, 2020).
```
Comment (iHiD) [inline reference/concepts/actor_model.md]: Should this have the `<>` removed now?

File: reference/concepts/actor_model.md [hunk 4]
```diff
@@ -1,7 +1,7 @@
 # Actor Model
 
-> The _actor model_ is a computational model of concurrency where an _actor_ is the computational primitive.  In response to messages sent by other _actors_, the _actor_ can: make local decisions, create more actors, send more messages, determine how to respond to the next message.  They can manage their own private state, and can affect other _actors_ only by sending a message.
+> The _actor model_ is a computational model of concurrency where an _actor_ is the computational primitive. In response to messages sent by other _actors_, the _actor_ can: make local decisions, create more actors, send more messages, determine how to respond to the next message. They can manage their own private state, and can affect other _actors_ only by sending a message.<sup>1</sup>
```
Comment (SaschaMann) [inline reference/concepts/actor_model.md]: If we agree to refer to sources for individual parts of the document with `<sup>n</sup>`, I think it should be documented in the FAQ.

File: reference/concepts/actor_model.md [hunk 5]
```diff
@@ -1,7 +1,7 @@
...
 
-## Sources
+---
```
Comment (SaschaMann) [inline reference/concepts/actor_model.md]: Could we add this to the FAQ as well, to standardise on a format?

File: reference/concepts/actor_model.md (hunk 3 above)
Comment (ErikSchierboom) [reply reference/concepts/actor_model.md]: Yep!

File: reference/concepts/actor_model.md (hunk 5 above)
Comment (ErikSchierboom) [reply reference/concepts/actor_model.md]: Done!

File: reference/concepts/actor_model.md (hunk 4 above)
Comment (ErikSchierboom) [reply reference/concepts/actor_model.md]: Done!

File: docs/maintainers/faqs.md [hunk 6]
```diff
@@ -131,6 +131,22 @@ Short answer: no. For the moment, we have all the v3 tracks in this single monor
...
+
+Yes! Please use citations in any reference documentation (files under `**/reference/**`). We are using the [Science of Computer Programming citation style][socp-citation-style]. This means that when citing website, please use this format:
+
+> $TITLE, $SOURCE. ($PUBLISHED_DATE). $URL (accessed \$ACCESSED_DATE).
+
+For example:
+
+> [1] Actor model, Wikipedia. (2020). https://en.wikipedia.org/w/index.php?title=Actor_model&oldid=939106706 (accessed February 29, 2020).
+
+To include a citation in reference documentation, add the citation reference to the bottom of the page. Then add with a line with the `---` separator above the citation. Finally, put the cited text in [blockquotes][blockquotes] and append a superscript referring to the citation number (usually `<sup>1</sup>`). See the [actor model reference documentation][actor_model-reference-documentation] for an example.
```
Comment (SaschaMann) [inline docs/maintainers/faqs.md]: > Finally, put the cited text in [blockquotes][blockquotes] 

That is only true for direct quotes, not for documents that are based on or derived from other sources that you'd still want to cite, isn't it?

For example in the macro and metaprogramming files below, the quoted text is *not* a direct quote and shouldn't be marked as such.

Comment (ErikSchierboom) [reply docs/maintainers/faqs.md]: Good catch. I'll update.

**Comment by ErikSchierboom** [comment]
Thanks everybody!

File: docs/maintainers/faqs.md [hunk 7]
```diff
@@ -131,6 +131,24 @@ Short answer: no. For the moment, we have all the v3 tracks in this single monor
...
+
+> [1] Actor model, Wikipedia. (2020). https://en.wikipedia.org/w/index.php?title=Actor_model&oldid=939106706 (accessed February 29, 2020).
+
+To include a citation in reference documentation, add the citation reference to the bottom of the page. Then add a line with the `---` separator above the citation. The final step is to append a superscript to the cited text. This superscript should refer to the citation reference's number (usually `<sup>1</sup>`). See the [actor model reference documentation][actor_model-reference-documentation] for an example.
+
+Note that if the cited text is copied literally, please put it in [blockquotes][blockquotes].
+
+### Generating citations
+
+To help generate these citations, [ZoteroBib][zoterobib] can be used. On that website, first set the citation style to "Science of Computer Programming" in the "Bibliography" section. Then, paste the URL you want to cite in the input field at the top of the page, click on the "Cite" button and the citation will be generated for you.
```
Comment (SleeplessByte) [inline docs/maintainers/faqs.md]: Select citation format dropdown


Select citation format search box

Maybe we want to include these?

**Comment by iHiD** [comment]
@SleeplessByte Yes please!

File: docs/maintainers/faqs.md (hunk 7 above)
Comment (ErikSchierboom) [reply docs/maintainers/faqs.md]: You mean the fact that you have to add it yourself?

Comment (iHiD) [reply docs/maintainers/faqs.md]: I think he's suggesting we add screenshots to the FAQ

**Pull request merged**